diff --git a/src/pacman/meson.build b/src/pacman/meson.build
index 67f34494..6926f676 100644
--- a/src/pacman/meson.build
+++ b/src/pacman/meson.build
@@ -13,7 +13,6 @@ pacman_sources = files('''
   callback.h callback.c
   upgrade.c
   util.h util.c
-  update-notice.h update-notice.c
 '''.split())
 
 pacman_conf_sources = files('''
diff --git a/src/pacman/sync.c b/src/pacman/sync.c
index 36dbc8ee..88f4a876 100644
--- a/src/pacman/sync.c
+++ b/src/pacman/sync.c
@@ -37,7 +37,7 @@
 #include "package.h"
 #include "callback.h"
 #include "conf.h"
-#include "update-notice.h"
+//#include "update-notice.h"
 
 static int unlink_verbose(const char *pathname, int ignore_missing)
 {
@@ -719,10 +719,10 @@ static int sync_trans(alpm_list_t *targets)
 			alpm_logaction(config->handle, PACMAN_CALLER_PREFIX,
 					"starting full system upgrade\n");
 		}
-		if (do_update_notice() == -1) {
-			trans_release();
-			return 1;
-		}
+		//if (do_update_notice() == -1) {
+		//	trans_release();
+		//	return 1;
+		//}
 
 		if(alpm_sync_sysupgrade(config->handle, config->op_s_upgrade >= 2) == -1) {
 			pm_printf(ALPM_LOG_ERROR, "%s\n", alpm_strerror(alpm_errno(config->handle)));
