From 4b65a23c80e70db03b9646bf038a9e2508e367f0 Mon Sep 17 00:00:00 2001
From: Mihai Fufezan <fufexan@protonmail.com>
Date: Thu, 4 Jan 2024 01:19:06 +0200
Subject: [PATCH] Nix: build using CMake

---
 nix/default.nix | 31 +++++++++++++------------------
 1 file changed, 13 insertions(+), 18 deletions(-)

diff --git a/nix/default.nix b/nix/default.nix
index f960f11..7ecbc28 100644
--- a/nix/default.nix
+++ b/nix/default.nix
@@ -1,55 +1,50 @@
 {
   lib,
   stdenv,
+  cmake,
   makeWrapper,
-  meson,
-  ninja,
   pkg-config,
+  wayland-scanner,
+  wrapQtAppsHook,
+  hyprland,
+  hyprland-protocols,
   hyprlang,
   libdrm,
   mesa,
   pipewire,
-  sdbus-cpp,
-  systemd,
-  wayland-protocols,
-  wayland-scanner,
   qtbase,
   qttools,
   qtwayland,
-  wrapQtAppsHook,
-  hyprland,
+  sdbus-cpp,
   slurp,
-  hyprland-protocols,
+  systemd,
   wayland,
+  wayland-protocols,
   debug ? false,
   version ? "git",
 }:
 stdenv.mkDerivation {
-  pname = "xdg-desktop-portal-hyprland";
+  pname = "xdg-desktop-portal-hyprland" + lib.optionalString debug "-debug";
   inherit version;
 
   src = ../.;
 
-  mesonBuildType =
-    if debug
-    then "debug"
-    else "release";
+  cmakeFlags = lib.optional debug (lib.strings.cmakeFeature "CMAKE_BUILD_TYPE" "DEBUG");
 
   nativeBuildInputs = [
-    meson
-    ninja
+    cmake
+    makeWrapper
     pkg-config
     wayland-scanner
-    makeWrapper
     wrapQtAppsHook
   ];
 
   buildInputs = [
     hyprland-protocols
+    hyprlang
     libdrm
     mesa
     pipewire
-    hyprlang
     qtbase
     qttools
     qtwayland
