--- numpy-1.24.0/numpy/core/src/umath/loops_unary_fp.dispatch.c.src~	2022-12-18 03:38:35.000000000 +0000
+++ numpy-1.24.0/numpy/core/src/umath/loops_unary_fp.dispatch.c.src	2023-01-20 21:24:45.632231249 +0000
@@ -11,7 +11,6 @@
  * to handle non-contiguous memory access comparing with SSE for
  * such small operations that this file covers.
 */
-#define NPY_SIMD_FORCE_128
 #include "numpy/npy_math.h"
 #include "simd/simd.h"
 #include "loops_utils.h"
--- numpy-1.24.0/numpy/core/src/umath/loops_unary_fp.dispatch.c.src~	2023-01-20 21:24:45.000000000 +0000
+++ numpy-1.24.0/numpy/core/src/umath/loops_unary_fp.dispatch.c.src	2023-01-20 21:25:42.284802822 +0000
@@ -1,6 +1,6 @@
 /*@targets
  ** $maxopt baseline
- ** sse2 sse41
+ ** sse2 sse41 avx2
  ** vsx2
  ** neon asimd
  ** vx vxe
